Fix path to watchSources in watch.coffee

canavandl opened the pull request:
issues: fixes #2928 

Corrects path to watchSources in watch.coffee

damianavila commented:
LGTM, pinging @mattpap for another view before merging...

mattpap commented on bokehjs/gulp/tasks/watch.coffee:
```diff
@@ -5,7 +5,7 @@ runSequence = require "run-sequence"
-  gulp.watch "#{paths.coffee.watchSources}", ->
+  gulp.watch "#{paths.bokehjs.coffee.watchSources}", ->
     runSequence("scripts:build")
   gulp.watch "#{paths.css.watchSources}", ->
```
This needs to be fixed as well. Should be `less`.

mattpap commented:
I think that `{bokehjs,widgets}.coffee.watchSources` should be moved to `coffee.watchSources` and then `bokehjs` part removed from the appropriate task. Otherwise, `widgets.coffee.watchSources` is not going to be ever used.

mattpap commented:
As we attempt to fix things, this whole `watch` task's implementation is suspicious to me, because you have `runSequence()` twice, i.e. two streams, and gulp allows only one. Otherwise, you have to use stream merging or callbacks. I doubt this code does what it was intended to do. Perhaps no one noticed, because this task is rarely used and it effectively watches only coffeescript sources anyway (see my earlier comment).

canavandl commented:
Three things:

1) Per @mattpap, I refactored the paths.coffee object to the following. I think it's a lot better.

```coffeescript
  coffee:
    bokehjs:
      destination:
        ...
    widgets:
      destination:
        ...
    sources: [
        "./src/coffee/main.coffee"
        "./src/coffee/widget/main.coffee"
    ]
    watchSources: [
      "./src/coffee/**/**"
    ]
```

2) I changed the gulp `watch` task to use the less files as the watchSources (not CSS), so that should work now too.

3) About the overall `watch` task implementation, it appears (to me) that runSequence is sent as a callback to `gulp.watch` so it's able to build CSS/JS concurrently. It "works" for me, but I'm no gulp expert.

bryevdv commented:
@brendancol @mistakevin @mattpap thoughts? I don't have context/opinions here.

mattpap commented:
I will have to make sure what this `runSequence()` actually does, because it doesn't fit into any of patterns that I know gulp tasks support. Anyway, we can merge this PR as-is and investigate this later.
